crapssim/strategy/single_bet.py

```diff
@@ -58,21 +60,26 @@ def update_bets(self, player: Player) -> None:
         if not self.bet.is_allowed(player):
             return
 
-        if self.mode == StrategyMode.ADD_IF_NON_EXISTENT:
-            BetIfTrue(self.bet, lambda p: self.bet not in p.bets).update_bets(player)
-        elif self.mode == StrategyMode.ADD_IF_POINT_ON:
-            BetPointOn(self.bet).update_bets(player)
-        elif self.mode == StrategyMode.ADD_IF_POINT_OFF:
-            BetPointOff(self.bet).update_bets(player)
-        elif self.mode == StrategyMode.ADD_IF_NEW_SHOOTER:
-            BetNewShooter(self.bet).update_bets(player)
-        elif self.mode == StrategyMode.ADD_OR_INCREASE:
-            player.add_bet(self.bet)
-        elif self.mode == StrategyMode.REPLACE:
-            existing_bets = player.already_placed_bets(self.bet)
-            for bet in existing_bets:
-                player.remove_bet(bet)
-            player.add_bet(self.bet)
+        match self.mode:
+            case StrategyMode.ADD_IF_NOT_BET:
+                AddIfNotBet(self.bet).update_bets(player)
+            case StrategyMode.ADD_IF_POINT_ON:
+                AddIfPointOn(self.bet).update_bets(player)
+            case StrategyMode.ADD_IF_POINT_OFF:
+                AddIfPointOff(self.bet).update_bets(player)
+            case StrategyMode.ADD_IF_NEW_SHOOTER:
+                AddIfNewShooter(self.bet).update_bets(player)
+            case StrategyMode.ADD_OR_INCREASE:
+                player.add_bet(self.bet)
+            case StrategyMode.BET_IF_POINT_ON:
+                AddIfPointOn(self.bet).update_bets(player)
+                # If only betting when point on, also need to turn off when point off
+                RemoveIfPointOff(self.bet).update_bets(player)
```

Comment:
New strategy mode BET_IF_POINT_ON needs to also RemoveIfPointOff for the input bet